frontend/utils/lazyLoadedView.js

```diff
@@ -1,16 +1,35 @@
 import Vue from 'vue'
-// NOTE: any modals opened with `OPEN_MODAL` go here
-Vue.component('LoginModal', () => import('../views/containers/access/LoginModal.vue'))
-Vue.component('SignupModal', () => import('../views/containers/access/SignupModal.vue'))
-Vue.component('UserSettingsModal', () => import('../views/containers/user-settings/UserSettingsModal.vue'))
-Vue.component('PasswordModal', () => import('../views/containers/access/PasswordModal.vue'))
-Vue.component('GroupLeaveModal', () => import('../views/containers/group-settings/GroupLeaveModal.vue'))
-// Vue.component('GroupDeletionModal', () => import('../views/containers/group-settings/GroupDeletionModal.vue'))
-Vue.component('GroupMembersAllModal', () => import('../views/containers/dashboard/GroupMembersAllModal.vue'))
-Vue.component('InvitationLinkModal', () => import('../views/containers/group-settings/InvitationLinkModal.vue'))
-
-Vue.component('GroupCreationModal', () => import('../views/containers/group-settings/GroupCreationModal.vue'))
-Vue.component('GroupJoinModal', () => import('../views/containers/group-settings/GroupJoinModal.vue'))
+import LoadingPage from '@views/containers/loading-error/LoadingPage.vue'
+import LoadingModal from '@views/containers/loading-error/LoadingModal.vue'
+import LoadingModalFullScreen from '@views/containers/loading-error/LoadingBaseModal.vue'
+
+// TODO Remove after design test period
+Vue.component('DSModalSimple', () => import('../views/containers/design-system/DSModalSimple.vue'))
+Vue.component('DSModalNested', () => import('../views/containers/design-system/DSModalNested.vue'))
+Vue.component('TimeTravel', () => import('../views/containers/navigation/TimeTravel.vue'))
+
+const lazyLoadView = ({ component, loading = LoadingPage, error = LoadingPage }) => {
+  const AsyncHandler = () => ({ component, loading, error })
+
+  return () =>
+    Promise.resolve({
+      functional: true,
+      render (h, { data, children }) {
+        return h(AsyncHandler, data, children)
+      }
+    })
+}
```

Comment:
In my local experiments, and from what I read of the Vue documentation, it is not necessary to have this function wrap the component inside of a functional component. So returning an immediately resolved promise isn't necessary, and the code below works just as well:

```js
const lazyLoadView = ({ component, loading = LoadingPage, error = LoadingPage }) => {
  return () => ({ component, loading, error })
}
```

However, additionally, neither the original version, nor the version above, showed me any loading screens/pages when I used Firefox's GPRS throttling mode in the Network tab:

<img width="162" alt="Screen Shot 2020-09-11 at 6 33 19 PM" src="https://user-images.githubusercontent.com/138706/92984216-7462f680-f45d-11ea-9218-720d8714619a.png">

Instead what it did was it simply waited to load a component (without showing a loading screen) and then displayed the final component.

Reply:
You could probably save a few characters and make the lines shorter by passing in the options as an options object, e.g.:

```js
const lazyLoadView = (component, { loading = LoadingPage, error = ErrorPage }) => {
  return () => ({ component, loading, error })
}

// ... and then ...

const GroupDashboard = lazyLoadView(import('@pages/GroupDashboard.vue'))
```

Reply:
... @taoeffect like we mention before, I don't think you can emulate network throttling when the file are already on your computer.
Concerning you refactoring, I have no idea what you're trying to do.
Maybe this vue-router page doc could help you to understand the bit code written here: https://github.com/vuejs/vue-router/pull/2140/files#diff-7d999265ce5b22152fdffee108ca6385